### PR TITLE
Feature/shinkwrap dependencies

### PR DESCRIPTION
**Pull request by harryi3t**

### CODE REVIEW

**Comment by harryi3t** [comment]
### This will now show any new module(s) added or any existing module(s) removed/updated
image

File: dependencies.txt [hunk 1]
```diff
@@ -0,0 +1,79 @@
+assert
```
Comment (shamasis) [inline dependencies.txt]: store it as JSON

File: lib/bundle/index.js [hunk 2]
```diff
@@ -1,7 +1,11 @@
 var _ = require('lodash'),
+    fs = require('fs'),
```
Comment (shamasis) [inline lib/bundle/index.js]: remove all fs stuff. we need to return dependencies and not write to file

File: lib/bundle/index.js [hunk 3]
```diff
@@ -91,7 +102,21 @@ Bundle = function (options) {
 
 _.assign(Bundle.prototype, /** @lends Bundle.prototype */ {
     compile: function (done) {
```
Comment (shamasis) [inline lib/bundle/index.js]: compile should not change!

add a new prototype function

File: test/system/dependencies.test.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+describe('dependencies', function () {
+    var fs = require('fs'),
+        fixedDependencies,
+        currentDependencies,
+        dependencyFilesError;
+
+    try {
+        fixedDependencies = fs.readFileSync('.tmp/dependencies.txt').toString().split('\n');
+        currentDependencies = fs.readFileSync('dependencies.txt').toString().split('\n');
```
Comment (shamasis) [inline test/system/dependencies.test.js]: store reference dependencies inside the test spec itself.

File: lib/bundle/index.js [hunk 5]
```diff
@@ -95,6 +95,27 @@ _.assign(Bundle.prototype, /** @lends Bundle.prototype */ {
     }
 });
 
+_.assign(Bundle.prototype, /** @lends Bundle */ {
+    listDependencies: function (done) {
+        var dependencies = [],
+            addPackageToDependencies = function (pck) {
```
Comment (shamasis) [inline lib/bundle/index.js]: pkg

**Comment by harryi3t** [comment]
## Listing all sandbox dependencies
image

## Searching for a particular dependency (will print all the places it's used with any version)
image

**Pull request merged**